### PR TITLE
Save file to bolb

### PR DESCRIPTION
**Pull request by ShaDy03**

### CODE REVIEW

File: IssueTracker/IssueTracker.FileSystem/Data/BaseClass.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace IssueTracker.FileSystem.Data;
+public class BaseClass
+{
+    protected string ConnectionString { get; set; }
+    protected string Container { get; set; } //TODO: aici o sa schimb treaba asta
+    public BaseClass(IConfiguration config)
+    {
+        ConnectionString = "AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;DefaultEndpointsProtocol=http;BlobEndpoint=http://127.0.0.1:10000/devstoreaccount1;" +
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/BaseClass.cs]: move into appsettings.json

File: IssueTracker/IssueTracker.FileSystem/Data/BaseClass.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace IssueTracker.FileSystem.Data;
+public class BaseClass
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/BaseClass.cs]: base for what? needs a better name

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IBolbConfiguration.cs [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace IssueTracker.FileSystem.Data.IData;
+public interface IBolbConfiguration
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/IData/IBolbConfiguration.cs]: rename to IBolbConfigurationFactory

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IBolbData.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿using Azure.Storage.Blobs;
+
+namespace IssueTracker.FileSystem.Data.IData;
+public interface IBolbData
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/IData/IBolbData.cs]: all methods should be async. 
the 'data' suffix is not appropriate for interfaces.

File: IssueTracker/IssueTracker.FileSystem/Data/BolbData.cs [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    public BolbData(IConfiguration config) : base(config)
+    {
+    }
+
+    public String Get(string name)
+    {
+        BlobServiceClient blobServiceClient = new BlobServiceClient(ConnectionString);
+        string containerName = Container;
+        BlobContainerClient containerClient = blobServiceClient.GetBlobContainerClient(containerName);
+        BlobClient x = containerClient.GetBlobClient(name);
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Data/BolbData.cs]: x? use a better name

File: IssueTracker/IssueTracker.FileSystem/Models/FileModel.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿namespace IssueTracker.FileSystem.Models;
+public class FileModel
+{
+#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider declaring as nullable.
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Models/FileModel.cs]: Do we need to disable nullable here?

File: IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IMetaDataProvider.cs [hunk 7]
```diff
@@ -2,10 +2,10 @@
 
 namespace IssueTracker.FileSystem;
 
-public interface IMetaDataProvider
+internal interface IMetaDataProvider
 {
     IEnumerable<MetaDataResponse> GetAll();
-    Task CreateAsync(MetaDataRequest entity);
+    Task CreateAsync(FileModel entity); //I change MetadataRequest in FileModel
```
Comment (iiusan) [inline IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IMetaDataProvider.cs:8]: Is this comment intended to be merged into the main branch?

File: IssueTracker/IssueTrackerAPI/Controllers/FileController.cs [hunk 8]
```diff
@@ -1,44 +1,40 @@
...
     }
 
-    [HttpGet("getFile")]
-    public IActionResult GetFileBlob(string fileName)
-    {
-        var sasLink = _bolbData.Get(fileName);
-        if (sasLink == null) throw new ArgumentException("sasLink is null!");
-        return Ok(sasLink); //TODO: nu ramane asa, o sa revin azi peste tot ce am implementat pentru a continua
-    }
+    //[HttpGet("getFiles")]
```
Comment (iiusan) [inline IssueTracker/IssueTrackerAPI/Controllers/FileController.cs]: keep the code clean

File: IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IMetaDataProvider.cs (hunk 7 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Repository/IRepository/IMetaDataProvider.cs:8]: no, this comment will be deleted

File: IssueTracker/IssueTrackerAPI/Controllers/FileController.cs (hunk 8 above)
Comment (ShaDy03) [reply IssueTracker/IssueTrackerAPI/Controllers/FileController.cs]: ok

File: IssueTracker/IssueTracker.FileSystem/Data/BaseClass.cs (hunk 1 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Data/BaseClass.cs]: image

File: IssueTracker/IssueTracker.FileSystem/Data/BaseClass.cs (hunk 2 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Data/BaseClass.cs]: i was deleted this class

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IBolbConfiguration.cs (hunk 3 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Data/IData/IBolbConfiguration.cs]: image

File: IssueTracker/IssueTracker.FileSystem/Data/IData/IBolbData.cs (hunk 4 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Data/IData/IBolbData.cs]: image
all methods are task
About name, IBolbFile? it's ok?

File: IssueTracker/IssueTracker.FileSystem/Data/BolbData.cs (hunk 5 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Data/BolbData.cs]: x change in blobClient

File: IssueTracker/IssueTracker.FileSystem/Models/FileModel.cs (hunk 6 above)
Comment (ShaDy03) [reply IssueTracker/IssueTracker.FileSystem/Models/FileModel.cs]: image

**Pull request merged**